packages/material-ui/test/typescript/hoc-interop.spec.tsx

```diff
@@ -9,44 +9,55 @@
  * See https://github.com/Microsoft/TypeScript/issues/28339 for in-depth discussion
  */
 import { Button, withStyles, createStyles } from '@material-ui/core';
-import TextField, { TextFieldProps } from '@material-ui/core/TextField';
+import TextField, { TextFieldVariant, TextFieldProps } from '@material-ui/core/TextField';
 import emotionStyled from '@emotion/styled';
 import * as React from 'react';
 import { RouteComponentProps, withRouter } from 'react-router';
 import styled from 'styled-components';
 
-const filledProps = {
-  InputProps: { classes: { inputAdornedStart: 'adorned' } },
+const outlineProps = {
+  InputProps: { classes: { notchedOutline: 'notchedOutline' } },
 };
 
-// baseline behavvior
-<TextField variant="filled" {...filledProps} />;
+// default behavior for standard variant
+<TextField InputProps={{ classes: { root: 'root' } }} />;
+
+// baseline behavior
+<TextField variant="outlined" {...outlineProps} />;
+
+// $ExpectError
+<TextField variant="filled" {...outlineProps} />; // desired to throw
 // $ExpectError
-<TextField {...filledProps} />; // desired to throw
+<TextField {...outlineProps} />;
 
 // styled
 {
   const StyledTextField = styled(TextField)``;
-  <StyledTextField variant="filled" {...filledProps} />; // desired to pass
-  <StyledTextField {...filledProps} />; // undesired, should throw
+  <StyledTextField variant="outlined" {...outlineProps} />; // desired to pass
```

Comment:
Could you add it instead of replacing the existing one? It's unclear whether the old test still passes.

Reply:
The main problem is that all props available for `filled` also available for `outlined`. Oh wait, there is `underlined` value. Ok